index.js

```diff
@@ -120,55 +120,57 @@ function checkForMatchInMessage(messageContent, matchingPattern) {
 }
 
 async function getMessageInfo(message) {
-    const result = await message;
+    const result = isPromise(message) ? await message : message;
     const messageDate = result.createdAt.toLocaleString();
     const messageAuthorId = result.author.id
-
     return {
         messageDate: messageDate,
         messageAuthorId: messageAuthorId
     }
 }
 
-async function checkPinsForMessageTracker(pins, generalChat) {
-    let messageTrackerExists = false; 
+async function getPins() {
+    const generalChat = getTargetChannel(process.env.GENERAL_CHAT_ID);
+    const pins = await generalChat.messages.fetchPinned();
+    return pins;
+}
 
+async function checkPinsForMessageTracker(pins) {
     for (const msg of pins.values()) {
         const messageTrackerMatch = checkForMatchInMessage(msg.content, MESSAGE_TRACKER_REGEX);
-        console.log(`Herre's messageTrackerMatch: ${messageTrackerMatch}`);
         if (messageTrackerMatch && messageTrackerMatch[0]) {
-            messageTrackerExists = true;
-            const messageTrackerId = msg.id;
-            console.log(`WE'RE IN`);
-            const messageTracker = await generalChat.messages.fetch(messageTrackerId);
-            console.log(`Here's messageTracker right after the await in the function: ${messageTracker}`);
-
-            return {    
-                messageTrackerExists: messageTrackerExists, 
-                messageTracker: messageTracker
-            };
+            MESSAGE_TRACKER_ID = msg.id;
+            return true;
         }
     };
+    return false;
+}
 
-    return {
-        messageTrackerExists: messageTrackerExists, 
-        messageTracker: null
-    };
+function checkForMessageTracker() {
```

Comment:
A quick two-minute challenge for you: could we turn this into a single line? 🤔
Additional challenge: try turning it into a single line without using the ternary operator 😉